staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -605,10 +605,10 @@ func (r *Request) tryThrottleWithInfo(ctx context.Context, retryInfo string) err
 	now := time.Now()
 
 	err := r.rateLimiter.Wait(ctx)
+	latency := time.Since(now)
 	if err != nil {
-		err = fmt.Errorf("client rate limiter Wait returned an error: %w", err)
+		err = fmt.Errorf("client rate limiter Wait returned an error after %v: %w", latency, err)
```

Comment:
> I'm pretty sure this only shows the time waiting since the last retry attempt

this is only called the first time the request is tried, inside `tryThrotltle`

interestingly the context.WithTimeout is added after it on normal requests

https://github.com/kubernetes/kubernetes/blob/564f43889209af368ed497053b7f2fe56c7c528e/staging/src/k8s.io/client-go/rest/request.go#L944-L955

and only used for Stream

https://github.com/kubernetes/kubernetes/blob/564f43889209af368ed497053b7f2fe56c7c528e/staging/src/k8s.io/client-go/rest/request.go#L811-L816

which context is canceling this request?